src/mcl.jl

```diff
@@ -0,0 +1,214 @@
+# MCL (Markov CLustering algorithm)
+
+#### Interface
+
+"""
+    `mcl()` result.
```

Comment:
I would rather indicate that it's immutable on the first line, than write it's a type (the latter is a little bit like explicitly writing "function that <...>" when documenting a function).

I think documenting the fields out-of-line and keeping both places in sync is too much of a burden, I would rather use some docstring extension that automatically syncs inline comments. Anyway, the fields should not be accessed directly in most of the cases, the user should use `ClusteringResult` API.